src/components/AddTodoForm/AddTodoForm.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+
+type Props = {
+  addNewTodo: (title: string) => void;
+};
+
+export const AddTodoForm:React.FC<Props> = ({
+  addNewTodo,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const handleInput = (event:React.ChangeEvent<HTMLInputElement>) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const resetForm = () => {
+    setTodoTitle('');
+  };
+
+  const handleSubmit = (event:React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasTitleError(!todoTitle);
+
+    if (!hasTitleError) {
+      addNewTodo(todoTitle);
```

Comment:
should we show some loader with temp todo here? for now as user I don't see that something is processing after form submitting